ldapadmin/src/main/java/org/georchestra/ldapadmin/ws/HomeController.java

```diff
@@ -19,66 +24,96 @@
-				redirectUrl = "/privateui/";
-			}
-			else {
-				redirectUrl = "/account/userdetails";
-			}
-			if(LOG.isDebugEnabled()){
-				LOG.debug("root page request -> redirection to " +
-						config.getPublicContextPath() + redirectUrl);
-			}
-			response.sendRedirect(config.getPublicContextPath() + redirectUrl);
-		} else {
-			response.setStatus(HttpServletResponse.SC_FORBIDDEN);
-			return;
-		}
-	}
-
-	@RequestMapping(value="/privateui/")
-	public String privateui(HttpServletRequest request) throws IOException{
-		String roles = request.getHeader("sec-roles");
-		if(roles != null && !roles.equals("ROLE_ANONYMOUS")) {
-			List<String> rolesList = Arrays.asList(roles.split(";"));
-			if(rolesList.contains("ROLE_MOD_LDAPADMIN")) {
-				return "privateUi";
-			}
-		}
-		return "forbidden";
-	}
+    private static final Log LOG = LogFactory.getLog(HomeController.class.getName());
+    private ExpiredTokenManagement tokenManagement;
+
+    private Configuration config;
+
+    @Autowired
+    private GeorchestraConfiguration georConfig;
+
+    @Autowired
+    private ServletContext context;
+
+    @Autowired
+    public HomeController(ExpiredTokenManagement tokenManagment, Configuration cfg) {
+        if(LOG.isDebugEnabled()){
+          LOG.debug("home controller initialization");
+        }
+        this.config = cfg;
+
+        this.tokenManagement = tokenManagment;
+        this.tokenManagement.start();
+    }
+
+    @RequestMapping(value = "/")
+    public void root(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        String roles = request.getHeader("sec-roles");
+
+        if (roles != null && !roles.equals("ROLE_ANONYMOUS")) {
+            String redirectUrl;
+            List<String> rolesList = Arrays.asList(roles.split(";"));
+
+            if (rolesList.contains("ROLE_MOD_LDAPADMIN")) {
+                redirectUrl = "/privateui/";
+            } else {
+                redirectUrl = "/account/userdetails";
+            }
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("root page request -> redirection to " + config.getPublicContextPath() + redirectUrl);
+            }
+            response.sendRedirect(config.getPublicContextPath() + redirectUrl);
+        } else {
+            // redirect to CAS
+            response.sendRedirect(config.getPublicContextPath() + "/account/userdetails?login");
+            return;
+        }
+    }
+
+    @RequestMapping(value="/privateui/")
+    public String privateui(HttpServletRequest request) throws IOException{
+        String roles = request.getHeader("sec-roles");
+        if(roles != null && !roles.equals("ROLE_ANONYMOUS")) {
+            List<String> rolesList = Arrays.asList(roles.split(";"));
+            if(rolesList.contains("ROLE_MOD_LDAPADMIN")) {
+                return "privateUi";
+            }
+        }
+        return "forbidden";
+    }
+
+    @RequestMapping(value = "/privateui/css/main.css")
```

Comment:
Both are customized at compilation time in the current build system (see https://github.com/georchestra/georchestra/tree/15.06/ldapadmin/src/main/filtered-resources/privateui). No other choice than exposing them via a specific controller if we want to let the admin customize it into the georchestra datadir.